cmd/internal/kube/inject.go

```diff
@@ -62,24 +79,36 @@ var injectCmd = &cobra.Command{
 		// Generate a secret for each namespace in the deployment if the user specified secret generation
 		secretBuf := new(bytes.Buffer)
 		if secretOpts.ShouldInject {
-			key, secret, err := cmderr.RequireAPICredentials("API credentials are required to generate secret.")
-			if err != nil {
-				return err
-			}
-
 			namespaces, err := injectr.InjectableNamespaces()
 			if err != nil {
 				return err
 			}
 
-			for _, namespace := range namespaces {
-				r, err := handleSecretGeneration(namespace, key, secret)
+			if postmanCollectionID != "" {
+				for _, namespace := range namespaces {
+					r, err := handlePostmanSecretGeneration(namespace, postmanAPIKey)
```

Comment:
I don't see any check that the postmanAPIKey is set (like there is for the Akita credentials below?)

Reply:
Since we will be getting it as the arguments itself, that's why I had not checked it here.
But since now we will be removing the Postman API key as argument, so will de the required changes.